src/hooks/useExchangeRate.js

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from 'react';
+
+export const useExchangeRate = () => {
+  const [exchangeRate, setExchangeRate] = useState(null);
+
+  useEffect(() => {
+    const fetchExchangeRate = async () => {
+      try {
+        const response = await Promise.all([
+          fetch('https://api.exchangerate-api.com/v4/latest/EUR'),
+        ]);
+
+        const [rateData] = await Promise.all(
+          response.map((response) => response.json()),
+        );
+
+        setExchangeRate(rateData.rates.PLN);
+      } catch (error) {
+        console.error('Error fetching exchange rate:', error);
+        // use default rate if api dont response
```

Comment:
Use Grammarly if you are not sure about the grammar

Reply:
<img width="742" height="227" alt="image" src="https://github.com/user-attachments/assets/b6701185-326d-4f1d-b2c5-7d0bbde05f2d" />